src/realm/group_shared.cpp

```diff
@@ -1860,6 +1859,8 @@ void SharedGroup::low_level_commit(uint_fast64_t new_version)
     out.set_versions(new_version, oldest_version);
     // Recursively write all changed arrays to end of file
     ref_type new_top_ref = out.write_group(); // Throws
+    m_free_space = out.get_free_space();
+    m_used_space = out.get_file_size() - m_free_space;
```

Comment:
We need to collect the stats as part of commit. Doing it outside of commit opens the result to interpretation (like: who was doing something to the file concurrently ... does the result even reflect the current state of the file?). Collecting the stats is of the same algorithmic complexity (but a much lower constant factor) as commit already has, since commit implies traversing the freelist looking for blocks to merge. So the relative cost is not high. It is constant.

If we should optimize it, we could add an argument to commit() which could control whether to collect the stats or not.

Summary: I think you are suggesting a premature optimization.